oaknational/Oak-Web-Application
# feat: posthog analytics

## Description

- adds posthog-js
- adds posthog apiKey and apiHost env/config
- adds Analytics context, inlcuding type safe tracking events
- adds tests
- adds `oakGlobals` which is intended to namespace any global variables we want to store at `window.__oakGlobals`
  

## Issue(s)

Fixes #304 

## How to test

1. Go to https://samara-ca79vdodd-oak-national-academy.vercel.app/
2. Should see in console logging for posthog events (including a custom track event)
3. Check in https://po8d5n.fra1.a.restack.io/events you should see events coming through like:
![image](https://user-images.githubusercontent.com/12934669/177824509-59cbbeaf-a4fd-420c-a55e-7485d09da286.png)


## Checklist
